include/oneapi/dpl/internal/distributed_ranges_impl/sp/algorithms/iota.hpp

```diff
@@ -0,0 +1,50 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _ONEDPL_DR_SP_IOTA_HPP
+#define _ONEDPL_DR_SP_IOTA_HPP
+
+#include <limits>
+
+#include "../../detail/std_ranges_shim.hpp"
+#include "../../views/iota.hpp"
+#include "for_each.hpp"
+
+namespace oneapi::dpl::experimental::dr::sp
+{
+
+template <distributed_range R, std::integral T>
+void
+iota(R&& r, T value)
+{
+    auto iota_view = stdrng::views::iota(value, T(value + stdrng::distance(r)));
+
+    for_each(par_unseq, views::zip(iota_view, r), [](auto&& elem) {
```

Comment:
It seems that for_each call with `par_unseq` leads to a static_assert within `oneapi::dpl::experimental::dr::sp::for_each` implementation....